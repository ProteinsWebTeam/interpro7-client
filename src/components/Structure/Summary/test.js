// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { SummaryStructure } from '.';

const renderer = new ShallowRenderer();

describe('<SummaryStructure />', () => {
  test('should render', () => {
    renderer.render(
      <SummaryStructure
        data={{
          loading: false,
          payload: {
            metadata: {
              accession: '102m',
              chains: ['A'],
              counters: {
                dbEntries: {
                  cathgene3d: 1,
                  interpro: 4,
                  panther: 1,
                  pfam: 1,
                  prints: 1,
                  profile: 1,
                  ssf: 1,
                },
                domain_architectures: 1,
                entries: 10,
                proteins: 1,
                proteomes: 1,
                sets: 1,
                taxa: 1,
              },
              experiment_type: 'x-ray',
              literature: {},
              name: {
                name: 'SPERM WHALE MYOGLOBIN H64A AQUOMET AT PH 9.0',
              },
              release_date: '1998-04-08T00:00:00Z',
              resolution: 1.84,
              source_database: 'pdb',
            },
          },
        }}
        dataMatches={{
          loading: false,
          payload: {
            count: 1,
            results: [
              {
                metadata: {
                  accession: 'G3DSA:1.10.490.10',
                  go_terms: null,
                  integrated: 'IPR012292',
                  member_databases: null,
                  name: 'Globins',
                  source_database: 'cathgene3d',
                  type: 'homologous_superfamily',
                },
                structures: [
                  {
                    accession: '102m',
                    chain: 'A',
                  },
                ],
              },
            ],
          },
        }}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
