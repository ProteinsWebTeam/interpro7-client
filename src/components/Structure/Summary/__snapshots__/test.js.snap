// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryStructure /> should render 1`] = `
<div
  className="sections"
>
  <section>
    <div
      className="row"
    >
      <div
        className="medium-9 columns margin-bottom-large"
      >
        <table
          className="light table-sum"
        >
          <tbody>
            <tr>
              <td>
                Accession
              </td>
              <td
                data-testid="structure-accession"
              >
                <Memo(Connect(Accession))
                  accession="102m"
                />
              </td>
            </tr>
            <tr>
              <td>
                Experiment type
              </td>
              <td
                className="text-cap"
                data-testid="structure-experiment-type"
              >
                <Memo(Connect(Link))
                  to={
                    Object {
                      "description": Object {
                        "entry": Object {
                          "db": "InterPro",
                          "isFilter": true,
                        },
                        "main": Object {
                          "key": "structure",
                        },
                        "structure": Object {
                          "db": "PDB",
                        },
                      },
                      "search": Object {
                        "experiment_type": "x-ray",
                      },
                    }
                  }
                >
                  X-ray
                </Memo(Connect(Link))>
              </td>
            </tr>
            <tr>
              <td>
                Resolution
              </td>
              <td
                data-testid="structure-resolution"
              >
                1.84
                 Ã…
              </td>
            </tr>
            <tr>
              <td>
                Chains
              </td>
              <td
                data-testid="structure-chains"
              >
                A
              </td>
            </tr>
            <tr>
              <td>
                Released
              </td>
              <td
                data-testid="structure-date"
              >
                <TimeAgo
                  date={1998-04-08T00:00:00.000Z}
                  noUpdate={true}
                  title="08/04/1998"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        className="medium-3 columns"
      >
        <div
          className="panel"
        >
          <h5>
            External Links
          </h5>
          <ul
            className="no-bullet"
            data-testid="structure-external-links"
          >
            <li>
              <BaseLink
                className="ext"
                id="102m"
                pattern="https://www.ebi.ac.uk/pdbe/entry/pdb/{id}"
                target="_blank"
              >
                PDBe
              </BaseLink>
            </li>
            <li>
              <BaseLink
                className="ext"
                id="102m"
                pattern="https://www.rcsb.org/structure/{id}"
                target="_blank"
              >
                RCSB PDB
              </BaseLink>
            </li>
            <li>
              <BaseLink
                className="ext"
                id="102m"
                pattern="http://www.ebi.ac.uk/thornton-srv/databases/cgi-bin/pdbsum/GetPage.pl?pdbcode={id}"
                target="_blank"
              >
                PDBsum
              </BaseLink>
            </li>
            <li>
              <BaseLink
                className="ext"
                id="102m"
                pattern="https://www.cathdb.info/pdb/{id}"
                target="_blank"
              >
                CATH
              </BaseLink>
            </li>
            <li>
              <BaseLink
                className="ext"
                id="102m"
                pattern="http://scop.mrc-lmb.cam.ac.uk/search?t=txt;q={id}"
                target="_blank"
              >
                SCOP
              </BaseLink>
            </li>
            <li>
              <BaseLink
                className="ext"
                id="102m"
                pattern="http://prodata.swmed.edu/ecod/complete/search?kw={id}"
                target="_blank"
              >
                ECOD
              </BaseLink>
            </li>
            <li>
              <BaseLink
                className="ext"
                id="102m"
                pattern="https://proteopedia.org/wiki/index.php/{id}"
                target="_blank"
              >
                Proteopedia
              </BaseLink>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <Memo(Connect(UnconnectedErrorBoundary))>
      <div
        className="row"
      >
        <div
          className="columns"
        >
          <Memo(Connect(StructureView))
            id="102m"
            matches={
              Array [
                Object {
                  "metadata": Object {
                    "accession": "G3DSA:1.10.490.10",
                    "go_terms": null,
                    "integrated": "IPR012292",
                    "member_databases": null,
                    "name": "Globins",
                    "source_database": "cathgene3d",
                    "type": "homologous_superfamily",
                  },
                  "structures": Array [
                    Object {
                      "accession": "102m",
                      "chain": "A",
                    },
                  ],
                },
              ]
            }
          />
        </div>
      </div>
    </Memo(Connect(UnconnectedErrorBoundary))>
  </section>
  <div />
</div>
`;
